modules/ref_device_module/src/ref_device_module_impl.cpp

```diff
@@ -55,9 +55,20 @@ DevicePtr RefDeviceModule::onCreateDevice(const StringPtr& connectionString,
         throw AlreadyExistsException();
     }
 
-    std::string localId = fmt::format("ref_dev{}", id);
+    const auto options = context.getOptions();
+    StringPtr localId;
+
+    if (options.assigned() && options.hasKey("ReferenceDevice"))
+    {
+        const DictPtr<StringPtr, StringPtr> referenceDevice = options.get("ReferenceDevice");
+        if (referenceDevice.hasKey("LocalId"))
+            localId = referenceDevice.get("LocalId");
+    }
```

Comment:
Could we add another field here called "Name" that would override the name of the device? 

I'd change the name of the device, as well as the name advertised in the services to "Reference device simulator" - but only in the simulator image.